src/components/MDXComponents/TokenList.tsx

```diff
@@ -0,0 +1,91 @@
+import type { Token } from 'prism-react-renderer';
+import type { TokenListProps } from './types';
+
+export const TokenList = ({
+  tokens,
+  showLineNumbers,
+  getLineProps,
+  getTokenProps
+}: TokenListProps) => {
+  let lineNumber = 0;
+  let shouldHighlight = false;
+  let highlightNextIndex: number | undefined;
+
+  return tokens.map((line: Token[], i: number) => {
+    let showLine = true;
+    lineNumber++;
+
+    // Test each line to see if it contains highlight comments. We
+    // have to build each token in the line array into a string because
+    // some languages (like graphQl) split comment text into different tokens.
+    let textLine = '';
+    for (let i = 0; i < line.length; i++) {
+      textLine += line[i].content;
+    }
```

Comment:
Do we have to worry about the `i` in this for loop and the `i` in the `tokens.map(...., i: number)` having a scope conflict or anything like that? 

I think changing the var name should be okay or we could do a `line.map(e => e.content).join('')`

Reply:
ah nevermind, I don't think there is a conflict actually, so it should be good to go!

Reply:
Oh, a .join() seems better/simpler than what i had so updated that. Thanks!